e2e/cypress/integration/post_header/post_header_spec.js

```diff
@@ -217,4 +217,21 @@ describe('Post Header', () => {
             cy.get(`#SEARCH_flagIcon_${postId}`).siblings('.post__pinned-badge').should('exist');
         });
     });
+
+    it('M17442 Visual verification of "Searching" animation for Flagged and Pinned posts', () => {
+        // # Go to Town-Square channel
+        cy.visit('/ad-1/channels/town-square');
+
+        // # Click on the "Pinned Posts" icon to the left of the "Search" box
+        cy.get('#channelHeaderPinButton').click();
+
+        // * Check before pinned posts are loaded, a "Searching..." animation appears on the top of the RHS
+        cy.get('#loadingSpinner').should('be.visible').and('have.text', 'Searching...');
```

Comment:
Please have this verification within the RHS. That could be:
```javascript
        // * Verify that the RHS for pinned posts is opened.
        cy.get('#searchContainer').should('be.visible').within(() => {
            // * Check that searching indicator appears before the pinned posts are loaded
            cy.get('#loadingSpinner').should('be.visible').and('have.text', 'Searching...');
            cy.get('#search-items-container').should('be.visible');

            // # Close the RHS
            cy.get('#searchResultsCloseButton').should('be.visible').click();
        });
```

Note: Should add new searchContainer ID [here](https://github.com/mattermost/mattermost-webapp/blob/master/components/search_results/search_results.jsx#L293).

Reply:
@saturninoabril I got this error message
<img width="1272" alt="Screen Shot 2019-10-28 at 11 51 27 PM" src="https://user-images.githubusercontent.com/52615235/67825229-fc871a00-fa9e-11e9-81b5-7c4fc4acebab.png">. I am not sure where to add the new searchContainer ID in the file you provided. Can you give me an example? Thanks!

Reply:
[In here](https://github.com/mattermost/mattermost-webapp/blob/e1a412d5262e01ff4fad67a7b65b102791ddc8bd/components/search_results/search_results.jsx#L304), you may change it as:
```diff
@@ -290,7 +290,10 @@ class SearchResults extends React.Component {
         }
 
         return (
-            <div className='sidebar-right__body'>
+            <div
+                id='searchContainer'
+                className='sidebar-right__body'
+            >
                 <SearchResultsHeader>
                     {formattedTitle}
                 </SearchResultsHeader>
@@ -327,4 +330,4 @@ class SearchResults extends React.Component {
     }
 }
```

Reply:
@saturninoabril I added the searchContainer ID to the file, but I am still getting the error message: "CypressError: Timed out retrying: Expected to find element: '#searchContainer', but never found it"
<img width="1267" alt="Screen Shot 2019-10-30 at 11 12 34 PM" src="https://user-images.githubusercontent.com/52615235/67916022-d3cc5680-fb6a-11e9-8f22-aa12b020e6c4.png">

Reply:
@saturninoabril nvm, I figured out now. Thanks!